### PR TITLE
Deprecate passing args as positional in DataFrame/Series.interpolate

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] xref #41485
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/series/methods/test_interpolate.py [hunk 1]
```diff
@@ -811,3 +811,13 @@ def test_interpolate_unsorted_index(self, ascending, expected_values):
...
+
+    def test_interpolate_pos_args_deprecation(self):
+        # https://github.com/pandas-dev/pandas/issues/41485
+        ser = Series([1, 2, 3])
+        msg = (
+            r"Starting with Pandas version 2\.0 all arguments of interpolate except "
+            r"for the argument 'self' will be keyword-only"
+        )
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            ser.interpolate(0)
```
Comment (simonjayhawkins) [inline pandas/tests/series/methods/test_interpolate.py:823]: I would have expected this to raise a TypeError.

File: pandas/tests/series/methods/test_interpolate.py [hunk 2]
```diff
@@ -811,3 +811,13 @@ def test_interpolate_unsorted_index(self, ascending, expected_values):
         result = ts.sort_index(ascending=ascending).interpolate(method="index")
         expected = Series(data=expected_values, index=expected_values, dtype=float)
         tm.assert_series_equal(result, expected)
+
+    def test_interpolate_pos_args_deprecation(self):
+        # https://github.com/pandas-dev/pandas/issues/41485
+        ser = Series([1, 2, 3])
+        msg = (
+            r"Starting with Pandas version 2\.0 all arguments of interpolate except "
+            r"for the argument 'self' will be keyword-only"
```
Comment (simonjayhawkins) [inline pandas/tests/series/methods/test_interpolate.py:820]: message doesn't need to mention self

**Comment by jorisvandenbossche** [comment]
Maybe allow `method` to be positional?

**Comment by jreback** [comment]
can you rebase

**Comment by jreback** [comment]
thanks @MarcoGorelli

**Comment by jreback** [comment]
@MarcoGorelli if you wouldn't mind adding these to the deprecations list see #30228 (I added this one).

also after these are 'done', it may make sense to make a separate deprecation section and/or normalize the language for these in the whatsnew.

File: pandas/core/series.py [hunk 3]
```diff
@@ -5256,6 +5257,29 @@ def to_period(self, freq=None, copy=True) -> Series:
...
+            method,
+            axis,
+            limit,
+            inplace,
+            limit_direction,
+            limit_area,
+            downcast,
+            **kwargs,
+        )
+
```
Comment (simonjayhawkins) [inline pandas/core/series.py:5282]: if deprecate_nonkeyword_arguments is updated to accept a class argument then maybe could do something like

```
    interpolate = deprecate_nonkeyword_arguments(
        version=None, allowed_args=["self", "method"], class="DataFrame"
    )(NDFrame.interpolate)
```

**Comment by MarcoGorelli** [comment]
will do - and yes, once they're in I'll sort out the whatsnew entry

**Pull request merged**